docs/en/tutorials/optim_wrapper.md

```diff
@@ -1,3 +1,500 @@
+
+model = nn.Linear(1, 1)
+optimizer = SGD(model.parameters(), lr=0.01)
+optim_wrapper = OptimWrapper(optimizer)
+
+print(optimizer.param_groups[0]['lr'])  # 0.01
+print(optimizer.param_groups[0]['momentum'])  # 0
+print(optim_wrapper.get_lr())  # {'lr': [0.01]}
+print(optim_wrapper.get_momentum())  # {'momentum': [0]}
+```
+
+```
+0.01
+0
+{'lr': [0.01]}
+{'momentum': [0]}
+```
+
+### Export/load state dicts
+
+Similar to the optimizer, the OptimWrapper provides the `state_dict` and `load_state_dict` interfaces for exporting and loading the optimizer states. For the `AmpOptimWrapper`, it can export mixed-precision training parameters as well.
+
+```python
+import torch.nn as nn
+from torch.optim import SGD
+from mmengine.optim import OptimWrapper, AmpOptimWrapper
+
+model = nn.Linear(1, 1)
+optimizer = SGD(model.parameters(), lr=0.01)
+
+optim_wrapper = OptimWrapper(optimizer=optimizer)
+amp_optim_wrapper = AmpOptimWrapper(optimizer=optimizer)
+
+# export state dicts
+optim_state_dict = optim_wrapper.state_dict()
+amp_optim_state_dict = amp_optim_wrapper.state_dict()
+
+print(optim_state_dict)
+print(amp_optim_state_dict)
+optim_wrapper_new = OptimWrapper(optimizer=optimizer)
+amp_optim_wrapper_new = AmpOptimWrapper(optimizer=optimizer)
+
+# load state dicts
+amp_optim_wrapper_new.load_state_dict(amp_optim_state_dict)
+optim_wrapper_new.load_state_dict(optim_state_dict)
+```
+
+```
+{'state': {}, 'param_groups': [{'lr': 0.01, 'momentum': 0, 'dampening': 0, 'weight_decay': 0, 'nesterov': False, 'maximize': False, 'foreach': None, 'params': [0, 1]}]}
+{'state': {}, 'param_groups': [{'lr': 0.01, 'momentum': 0, 'dampening': 0, 'weight_decay': 0, 'nesterov': False, 'maximize': False, 'foreach': None, 'params': [0, 1]}], 'loss_scaler': {'scale': 65536.0, 'growth_factor': 2.0, 'backoff_factor': 0.5, 'growth_interval': 2000, '_growth_tracker': 0}}
+```
+
+### Use multiple optimizers
+
+Considering that algorithms like GANs usually need to use multiple optimizers to train the generator and the discriminator, OptimWrapper provides a container class called `OptimWrapperDict` to manage. `OptimWrapperDict` stores the wrappers in the form of dicts and allows users to access and traverse the elements just like a dict.
+
+Unlike regular OptimWrapper, `OptimWrapperDict` does not provide methods such as `update_prarms`, `optim_context`, `backward`, `step`, etc. Plus, it cannot be used directly for training models. We suggest to access the elements in `OptimWrapperDict` directly to implement the parameter update logic.
+
+Users may wonder why not just use `dict` to manage multiple optimizers if `OptimWrapperDict` does not have training capabilities. Actually, the core function of `OptimWrapperDict` is to support batch exporting and loading of state dicts of all OptimWrappers, and to support getting learning rates and momentums as well. Without `OptimWrapperDict`, MMEngine needs to do a lot `if-else` in OptimWrapper to get all the states of the OptimWrappers.
```

Comment:
```suggestion
Users may wonder why not just use `dict` to manage multiple optimizers since `OptimWrapperDict` does not have training capabilities. Actually, the core function of `OptimWrapperDict` is to support exporting or loading the state dictionary of all sub-OptimWrapper and to support getting learning rates and momentums as well. Without `OptimWrapperDict`, MMEngine needs to do a lot of `if-else` in OptimWrapper to get the states of the `OptimWrappers`.
```